runner/master_log.py

```diff
@@ -95,7 +97,16 @@ def human_readable_full_log(self, test_id: str, message: str) -> List[str]:
                 sender = EVENT_SENDER[event.type]
                 messages.append(f"{sender} ({event.timestamp}): {event.data['message']}")
             elif event.type == EventType.WAIT:
-                messages.append(f"SYSTEM ({event.timestamp}): Test '{event.test_id}' WAITING for {event.data['tokens']} tokens until {event.data['time']}")
+
+                log_parts = [f"SYSTEM ({event.timestamp}): Test '{event.test_id}' WAITING for "]
+                if event.data['tokens'] > 0:
+                    log_parts.append(f"{event.data['tokens']} TOTAL TOKENS, ")
+                if event.data['time'] > datetime.now():
+                    log_parts.append(f"DATE {event.data['time']}, ")
+                if event.data['percentage_finished'] > 0.0:
+                    log_parts.append(f"{event.data['percentage_finished']}% of tests to be finished.")
+
+                messages.append("".join(log_parts))
```

Comment:
I meant the `", ".join(log_parts)` to avoid including the commas in the part 😅 but it's ok, just an aesthetic detail (e.g. if it's not waiting on percentage, the line will end in `, `).